---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->


```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse  = TRUE,
  comment   = "#>",
  fig.path  = "man/figures/",
  out.width = "100%"
)
```

# PAFTOL_magnoliids

<!-- badges: start -->
[![R CMD Check](https://github.com/ajhelmstetter/PAFTOL_magnoliids/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/ajhelmstetter/PAFTOL_magnoliids/actions/workflows/R-CMD-check.yaml)
[![License: GPL (>= 2)](https://img.shields.io/badge/License-GPL%20%28%3E%3D%202%29-blue.svg)](https://choosealicense.com/licenses/gpl-2.0/)
[![Dependencies](https://img.shields.io/badge/dependencies-16/144-red?style=flat)](#)
<!-- badges: end -->

Research Compendium of the project: PAFTOL Magnoliids

### Relevant programs

* Treeshrink: https://github.com/uym2/TreeShrink
* phyparts: https://bitbucket.org/blackrim/phyparts/wiki/Home
* discovista: https://github.com/esayyari/DiscoVista
* astral-pro: https://github.com/chaoszhang/A-pro
* c++ astral: https://github.com/chaoszhang/ASTER/tree/548fd31493397ec44f5c8e3452c689e547b3e271
* ggtree and plotly for interactive paralogs: https://yulab-smu.top/treedata-book/related-tools.html#plotly 

### Relevant papers

* Yan Z., Du P., Hahn M.W., Nakhleh L. 2020. Species tree inference under the multispecies coalescent on data with paralogs is accurate. bioRxiv:498378.

## Useful commands

## Comments on initial trees

Bill:

* Some jumbled Lauraceae/Monimiaceae

Thomas:

* Cananga is still posing problems, clustering with Endiandra far away from its normal position. I guess this sample is probably erroneous and should be removed.

EJdL:

* some strange positions for Lauraceae/Monimiaceae and Siparunaceae/Atherospermataceae
* Some of them I already saw in the tree generated by Alex for the Monimiaceae paper
	* Hortonia, Levieria montana, Austromatthaea elegans, and Decarydendron ranomafensis
	* I believe it is because they had few (or zero) genes recovered with the expected length
* I saw some genera missing in the tree
	* Matthaea, Pendressia and Monimia are missing for Monimiaceae
	* Sparattanthelium and Illigera for Hernandiaceae
	* Gomortega for Gomortegaceae
	* Chimonanthus, Idiospermum and Calycanthus for Calycanthaceae.

Zacky:

* The position of Haematodendron is obviously not correct, here as sister to Siparuna in both the first and second trees. The 75% tree shows it in a polytomy with Myristicaceae, so the family is still non-monophyletic. We should perhaps remove this sequence as it seems to be erroneous or otherwise problematic. Not sure choosing a different threshold will place it any better.
* Paramyristica is an interesting genus with an unstable position. It has never been placed properly before and it seems like that may end up being the case here as well. At least arguments of it possibly being including back into Myristica seem to now be debatable again.
* Bicuiba's position shifts in the 75% tree to become sister to the rest of the family, which is incorrect.
	* other genera also appear to be somewhat 'forced' into more incorrect positions in the 75% tree
	* For example, Osteophloeum which at 75% is sister to Virola which seems more incorrect than its position in the other two trees. I expect it to be sister to Iryanthera (similar to what Herve previously found), so I think either of its positions found here are questionable.
* I did not manage to see Doyleanthus - was that not included here?

Herve:

* Rooting: if there is a way of outputting future trees with Chloranthaceae as a clade, that would be great and bring us one step closer to publication outputs.
* We are missing a few entire families that we were supposed to have from 1KP an/or PAFTOL
	* Calycanthaceae (1KP and PAFTOL)
	* Gomortegaceae (1KP)
* We are missing most genera of Atherospermataceae (e.g., Atherosperma) and Lauraceae (e.g., Ocotea, and many more) 
	* Said to be sequenced by PAFTOL in sampling documents compiled by Olivier.
* Cananga odorata (PAFTOL), Endiandra glauca (PAFTOL or GAP) & Haematodendron glabrum (PAFTOL) are definitely suspicious and I wonder if this is a result of poor recovery for these species. Is there an easy way to output a summary table with basic stats to help us assess overall sequencing quality?
* Laurales: not sure what is going on with Siparunaceae, looks suspicious. 
	* Monimiaceae and Lauraceae totally mixed up: this makes no sense!
	* This issue did not occur in the preliminary trees sent by Andrew in December
	* Yhere are exactly four taxon names that appear to be swapped between the two families.
	* I suspect Austromatthaea elegans (GAP) is a distinct issue
* It would be critical to put an explicit species name on Virola sp. and Iryanthera sp.
* We will have to put explicit species names on those Annonaceae taxa collected together in Borneo.
* Append 1KP, PAFTOL, and GAP tags to the taxon names on all our trees
*  Is there any chance you could try a supermatrix analysis with either all exons or only those 28% with maximal overlap across orders (from Venn diagram)?
* Start assembling a publication-ready sampling table so we can check sample IDs if need be.
	* With species names, sources, vouchers, contributors, and project (1KP, PAFTOL, or GAP) will be extremely useful for us to continue the conversation
* Updated  metadata for the following sample IDs: 79724 to 79805
	* If you have any data from these sample ID ranges, please delete them as the metadata could be incorrect or the data is not usable (low quality).

Thomas part II:

* Just a quick point, the Uvariastrum pierreannum PAFTOL is in fact a mis identification.
* Just make a note of this: For Uvariastrum use this sample instead:
RUN79 Index 12 T48 (Uvariastrum zenkeri).
* Also you can delete one of the Uvariopsis you have, keep the best one I would say.
* You can also delete Boutiquea platypetala PAFTOL, it is now syn of Neostenanthera.

Roy : 

I also see some possibilities for deleting duplicated taxa, assuming we only need one per genus if the position stays fixed. I did not put them in the list yet, also here Thomas should give his opinion. Since I was not involved in the taxon selection the first time around, I do not want to make suggestions against things that might have been consciously been done. Options:

* Anaxagorea (either one)
* Drepananthus (either one)
* Tetramaranthus (either one)
* Pseuduvaria (keep PAFTOL?)
* Mithrephora (keep PAFTOL?)
* Xylopia (either one of the three)
* Annona (keep PAFTOL?)
* Uvariopsis (either one)
* Uvaria (keep PAFTOL?)

### To do: 

* Compare to PAFTOL 353 paper
* Alignments failed? CHECK: 5841, 5899, 6363
* hybpiper_stats.sh is crashing before the end, but produces results...?
* Rerun retrieve seqs including P_011527, P_011673 (already copied to hybpiper_magnoliids)
* Check metadata for Endiandra, Austromatthaea
* HS: confirm that you are indeed using the correct raw data file (e.g., by replicating the cleanup of that specific taxon)?
* If we too stringent, we may biasing the sampling towards taxa that are closer to the sequences represented in the bait kit

### How to cite

Please cite this compendium as: 

> **{{ CITATION TO BE ADDED UPON PUBLICATION }}**

### Content

This repository is structured as follow:

- [`data/`](https://github.com/ajhelmstetter/PAFTOL_magnoliids/tree/master/data):
contains all raw data required to perform analyses __(Hidden until publication)__

- [`rscripts/`](https://github.com/ajhelmstetter/PAFTOL_magnoliids/tree/master/rscripts/):
contains R scripts to run each step of the workflow

- [`outputs/`](https://github.com/ajhelmstetter/PAFTOL_magnoliids/tree/master/outputs):
contains all the results created during the workflow

- [`figures/`](https://github.com/ajhelmstetter/PAFTOL_magnoliids/tree/master/figures):
contains all the figures created during the workflow

- [`paper/`](https://github.com/ajhelmstetter/PAFTOL_magnoliids/tree/master/paper):
contains all the manuscript and related content (biblio, templates, etc.)

- [`R/`](https://github.com/ajhelmstetter/PAFTOL_magnoliids/tree/master/R):
contains R functions developed especially for this project

- [`man/`](https://github.com/ajhelmstetter/PAFTOL_magnoliids/tree/master/man):
contains help files of R functions

- [`DESCRIPTION`](https://github.com/ajhelmstetter/PAFTOL_magnoliids/tree/master/DESCRIPTION):
contains project metadata (author, date, dependencies, etc.)

- [`make.R`](https://github.com/ajhelmstetter/PAFTOL_magnoliids/tree/master/make.R):
master R script to run the entire project by calling each R script stored in the `rscripts/` folder

### Usage

Clone the repository, open the `.Rproj` file in RStudio and run:

```{r eval = FALSE}
source("make.R")
```

### Notes

- All required packages, listed in the `DESCRIPTION` file, will be installed (if necessary)
- All required packages and R functions will be loaded
- Some analyses listed in the `make.R` might take time

# Pipeline: data assembly

## GAP

### Download sequence data

Herve API key to download data:

```{bash apikey, eval = FALSE}
export CKAN_API_KEY=6a9c3f5d-b687-4ef9-8c9c-96d71cebf6f6
```

Sequence metadata:

[data/gap_metadata](https://github.com/ajhelmstetter/PAFTOL_magnoliids/blob/master/data/gap_metadata)

### Merge GAP data where samples have been sequenced multiple times

Shows R2 merging only (same script used for R1):

[scripts/combine_gap.sh](https://github.com/ajhelmstetter/PAFTOL_magnoliids/blob/master/bashscripts/combine_gap.sh)

## PAFTOL

### Get data from Kew sftp

```{bash ftp, eval = FALSE}
#connect to ftp
sftp anonymous@sftp.kew.org

#cd to directory
cd path/to/files

#download data to directory you sftp'd from
get *
```

### Filename change for the following files:

```{r rename_table, echo = FALSE}

r1s<-rbind(c("PAFTOL_014093_1.fastq.gz",	"PAFTOL_014095_1.fastq.gz"),
c("PAFTOL_014095_1.fastq.gz",	"PAFTOL_014097_1.fastq.gz"),
c("PAFTOL_014097_1.fastq.gz",	"PAFTOL_014099_1.fastq.gz"))

r1s<-data.frame(r1s)
colnames(r1s)<-c("Old R1 filename","New R1 filename")

knitr::kable(
  r1s, caption = 'Files that were renamed after naming error (suggested by O. Maurin).'
)
```

## hybpiper output

### Taxa with only 10 loci or fewer recovered

```{r low_qual_table, echo = FALSE}

lq<-rbind(c("Gbel","0"),
c("P_016915","0"),
c("P4172","0"),
c("S47","0"),
c("S24","3"),
c("S39","3"),
c("P0110D","4"),
c("P_014097","6"),
c("P_011589","8"),
c("S25","9"),
c("GAP_79789","10"),
c("P_007864","10"))

lq<-data.frame(lq)
colnames(lq)<-c("sample","no. exons recovered")

knitr::kable(
  lq, caption = 'Samples where we recovered a low number of exons using hybpiper'
)
```


## SRA

### Get paired-end fastqs after downloading from SRA

```{bash sra, eval = FALSE}
fastq-dump -I --split-files SRR10064004.1
```

## 1KP

K. Leempoel provided 353 recoveries from 1KP data.

## iqtree

```{bash iqtree, eval = FALSE}
iqtree -s raphia.fa -spp raphia.partitions -m GTR+I+G -nt AUTO -ntmax 16 -bb 1000
```

## ASTRAL

```{bash astral, eval = FALSE}

bash ~/scripts/post_genetrees_astral.sh ~/programs/ASTRAL/Astral.5.7.7/Astral/astral.5.7.7.jar astral_50_50

```

