###
# ASTRAL tree with hybpiper locus recovery
###

rm(list = ls())

library(ggplot2)
library(ggtree)
library(ape)
library(tidyr)
library(phylotools)

#read in tree and tips
phy <-read.tree("data/trees/astral-mafft_r10_l50_i50_b2_0/astral_bs10_LPP.tre")
df <- read.csv("data/sample_data - samples_for_phylo.csv")
df <- df[df$Namelist %in% phy$tip.label, ]

#NOTE: Not fixing branch lengths will cause plot to be cladogram, which is wanted
#fix terminal branches
#phy$edge.length[phy$edge.length == "NaN"] <- 0.25

#make all branch lengths the same
#phy$edge.length <- rep(0.1,length(phy$edge.length))

#reroot
phy <- ape::root(phy, node=251, edgelabel = T, resolve.root = T)

#ladderize
phy <- ladderize(phy)

#Hybpiper recovery dataset
#Here, set the file name of the table generated by get_seq_lengths.py
sample.filename = "data/test_seq_lengths.txt"

#use their script to import to ensure correct ref lengths
sample.data = as.matrix(read.table(
  sample.filename,
  header = T,
  row.names = 1,
  sep = "\t"
))

#remove X from locus name
colnames(sample.data) <- gsub("X", "", colnames(sample.data))

#set length of reference exon
reference.len = as.numeric(sample.data[1,])

#extract only sample exon lengths
sample.len = sample.data[2:length(sample.data[, 1]),]

#calculate percentage of exon recovered for each exon in each individual
percent.len = sweep(sample.len, 2, reference.len, "/")
percent.len = ifelse(percent.len > 1, 1, percent.len)

#drop 1kp
phy <- drop.tip(phy, setdiff(phy$tip.label, rownames(percent.len)))

#check diffs
setdiff(phy$tip.label, rownames(percent.len))
setdiff(rownames(percent.len), phy$tip.label)

rownames(percent.len) == phy$tip.label

#remove those in hybpiper but not in tree from table
percent.len <-
  percent.len[match(phy$tip.label, rownames(percent.len)), ]

#check diffs
setdiff(phy$tip.label, rownames(percent.len))
setdiff(phy$tip.label, rownames(percent.len))

rownames(percent.len) == phy$tip.label

#only include rows matching tips in the phylogenetic tree
df2 <- df[df$Namelist %in% phy$tip.label, ]
df2

#match the order of dataframe to tips
df2 <- df2[match(phy$tip.label, df2$Namelist), ]

#combine genus and species
df2 <-
  df2 %>% unite("genus_species",
               c(Genus:Species),
               na.rm = TRUE,
               remove = FALSE)

#tip labels
##tip label change
dfnames <- data.frame(df2$Namelist, df2$genus_species)
dfnames$df2.genus_species <- make.unique(dfnames$df2.genus_species)

#check
table(phy$tip.label==dfnames$df2.Namelist)

phy <- sub.taxa.label(phy, dfnames)

#change table names to match new tree names
rownames(percent.len) <- phy$tip.label

#melt table
library(reshape2)
percent.len_melt <- melt(percent.len)
head(percent.len_melt)
colnames(percent.len_melt) <- c("label", "category", "value")

percent.len_melt$label <- as.character(percent.len_melt$label)
percent.len_melt$category <- as.character(percent.len_melt$category)
head(percent.len_melt)
str(percent.len_melt)

##
# COLOUR BASED ON ORDER
##

#make a data frame containing columns for only the tip label and the corresponding order
df3 <- data.frame(phy$tip.label, df2$Order)
colnames(df3) <- c("label", "order")
head(df3)

#make sure there are no differences between data frame and phylogenetic tree
setdiff(phy$tip.label, df3$label)
setdiff(df3$label, phy$tip.label)

#tree
#if (!requireNamespace("BiocManager", quietly = TRUE))
#install.packages("BiocManager")
#BiocManager::install("ggtree")

#plot tree
library(ggtree)
g <- ggtree(phy, layout = "rectangular")
g
g2 <- g %<+% df3 +
  geom_tiplab(aes(color = factor(order)), size = 1.75) + theme(
    legend.position = c(0.1, 0.9),
    #legend.title = element_blank(),
    # no title
    #legend.key = element_blank(),
    legend.text = element_blank()) +
  guides(colour = guide_legend(override.aes = list(size = 5, label = sort(unique(df3$order))))) + labs(col="Order") +
  xlim(0,27) #set axis limits to put tree close to heatmap

g2

#CHECK IF THE DATA IS IN CORRECT ORDER
percent.len_melt$label[1:length(phy$tip.label)]==phy$tip.label

#plot heatmap
p2 <- ggplot(percent.len_melt, aes(x = category, y = label)) +
  geom_tile(aes(fill = value)) + scale_fill_viridis_c(name = "% exon length", option = "D") +
  theme_tree2() + theme(
    axis.line.x = element_blank(),
    axis.title.x = element_blank(),
    axis.ticks.x = element_blank(),
    axis.text.x = element_blank(),
    #axis.text.y=element_text(size=5), #uncomment to check table alignment
    #legend.title =element_blank(),
    legend.text = element_text(size = 7.5),
    legend.margin = margin(0, 0, 0, 0, "cm")
    #legend.position = "none"
  )

p2

pdf('figures/figS4_tree_exon_recovery.pdf',
    height = 20,
    width = 20)
library(aplot)
p2 %>% insert_left(g2,width=0.5)
dev.off()

ggsave("figures/figS4_tree_exon_recovery.png",height=40,width=40)
