---
title: "Magnoliids preliminary results"
output: pdf_document
fig_caption: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = '~/Dropbox/projects/AJH_magnoliids/PAFTOL_magnoliids')
```

```{r gradient, echo=F, message=F, warning=F,results='hide',fig.width=7,fig.height=5,fig.cap="A plot showing the number of exons recovered under a range of filtering thresholds based on two parameters. The x-axis shows first filtering parameter: the percentage length of the exon that was recovered. The second filtering parameter, on the y-axis, is the percentage of the total number of individuals for which an exon was recovered. A given combination of x and y values yields a number of exons kept after filtering at these parameter values. This number is represented by a colour gradient where deep red shades indicate more exons were kept after filtering. Grey cells are used when no exons were kept. White, dashed lines are contours that demonstrate the 3d structure of the data, or the rate of change of the number of exons recovered as filtering parameters are varied."}

library(ggpubr)
library(tidyverse)

#Here, set the file name of the table generated by get_seq_lengths.py
sample.filename = "data/test_seq_lengths.txt"

#use their script to import to ensure correct ref lengths
sample.data = as.matrix(read.table(
  sample.filename,
  header = T,
  row.names = 1,
  sep = "\t"
))

#remove X from locus name
colnames(sample.data) <- gsub("X", "", colnames(sample.data))

#set length of reference exon
reference.len = as.numeric(sample.data[1,])

#extract only sample exon lengths
sample.len = sample.data[2:length(sample.data[, 1]),]

#calculate percentage of exon recovered for each exon in each individual
percent.len = sweep(sample.len, 2, reference.len, "/")
percent.len = ifelse(percent.len > 1, 1, percent.len)

#read in taxonomy dataset
tn<-read.csv("data/sample_data - samples_for_phylo_AZ.csv")

#only those samples output put hybpiper
tn<-tn[tn$Include_hybpiper==1,]

#check for differences
#setdiff(tn$Namelist,rownames(percent.len))
#setdiff(rownames(percent.len),tn$Namelist)

#set thresholds
limits <- seq(0.01, 0.99, 0.01)

#make empty matrix
exon_stats <- matrix(nrow = length(limits), ncol = length(limits))

#loop through threshold combinations
for (i in 1:length(limits)) {
  for (j in 1:length(limits)) {

    #reset limits
    percent.len.limit <- percent.len

    # if percentage exon length recovered is >= 75% make value 1
    # if not make value 0
    percent.len.limit[percent.len.limit >= limits[i]] = 1
    percent.len.limit[percent.len.limit < limits[i]] = 0

    #For each exon calculate number of individuals with >= 75% of exon
    col <- colSums(percent.len.limit != 0)

    #Calculate % individuals with >= 75% of exon for each exon
    col <- col / (length(percent.len[, 1])) # -1 for meanlength

    exon_stats[i, j] <-  table(col >= limits[j])["TRUE"]

  }

}

#####
# Gradient plot
#####

#convert to data frame
exon_stats <- data.frame(exon_stats)

#convert to percentages
for (i in 1:length(limits)) {
  colnames(exon_stats)[i] <-  limits[i] * 100
  exon_stats$exon_prop[i] <-  limits[i] * 100
}

#melt table
library(reshape2)
exon_stats_melt <- melt(exon_stats, id.vars = "exon_prop")
exon_stats_melt$variable <- as.numeric(exon_stats_melt$variable)
colnames(exon_stats_melt) <- c("exon_prop", "ind_prop", "no_markers")

#gradient color
library(wesanderson)
pal <- wes_palette("Zissou1", 100, type = "continuous")

#gradient plot
p <- ggplot(exon_stats_melt, aes(exon_prop, ind_prop, z = no_markers))
p + geom_raster(aes(fill = no_markers)) +
  scale_fill_gradientn(colours = pal, name = "No. exons kept") +
  scale_y_continuous(expand = c(0, 0)) +
  scale_x_continuous(expand = c(0, 0)) +
  labs(y= "Filtering statistic 2: % individuals with exon", x = "Filtering statistic 1: % exon length recovered") +
  geom_contour(colour = "white", linetype = "dashed", alpha = 0.5) +
  theme(
    text = element_text(size = 10),
    axis.text = element_text(size = 10),
    axis.title.x = element_text(size = 11, margin = margin(
      t = 5,
      r = 0,
      b = 0,
      l = 0
    )),
    #axis.text.x=element_blank(),
    #axis.ticks.x=element_blank(),
    #axis.line.x = element_line(),
    #axis.title.y = element_blank(),
    #axis.text.y = element_blank(),
    #axis.ticks.y = element_blank(),
    #axis.line.y = element_blank(),
    legend.key = element_blank(),
    legend.title = element_text(),
    legend.text = element_text(size = 10),
    #legend.position = c(0.8, 0.9),
    panel.border = element_blank(),
    panel.grid.minor = element_blank(),
    panel.grid.major = element_blank(),
    panel.background = element_blank()
  )


```

```{r venn, echo=FALSE,message=F,warning=F,results='hide',fig.width=7,fig.height=7,fig.cap="Venn diagram showing overlap of exons in each order. Individuals were split by order into different datasets. Filtering was then done independently on each of the five datasets. Overlap among exons across orders was then calculated. This is shown as proportion of total exon dataset in each segment. Greyscale gradient represents the raw number of exons belonging to each segment."}

library(ape)

#Calculate sizes for gene and sample labels, increase the multiplier to make text bigger
gene.size.multiplier = .1
sample.size.multiplier = .5

sample.filename = "data/test_seq_lengths.txt"
sample.data= as.matrix(read.table(sample.filename,header=T,row.names=1,sep="\t"))

#length reference exon
reference.len = as.numeric(sample.data[1,])

#families
canel<-read.table("data/canel.txt")
canel<-as.character(canel$V1)

chlor<-read.table("data/chlor.txt")
chlor<-as.character(chlor$V1)

laura<-read.table("data/laura.txt")
laura<-as.character(laura$V1)

magno<-read.table("data/magno.txt")
magno<-as.character(magno$V1)

piper<-read.table("data/piper.txt")
piper<-as.character(piper$V1)

c(canel,chlor,laura,magno,piper)

ord_l<-list(canel,chlor,laura,magno,piper,c(canel,chlor,laura,magno,piper))

l_75_75<-list()

for(i in 1:6){

#extract only sample exon lengths
sample.len = sample.data[rownames(sample.data) %in% ord_l[[i]],]

#calculate percentage of exon recovered for each exon in each individual
percent.len= sweep(sample.len, 2, reference.len, "/")
percent.len = ifelse(percent.len>1,1,percent.len)
percent.len.limit <- percent.len

# if percentage exon length recovered is >= 75% make value 1
# if not make value 0
percent.len.limit[percent.len.limit>=0.5] = 1
percent.len.limit[percent.len.limit<0.5] = 0

#For each exon calculate number of individuals with >= 75% of exon
col <- colSums(percent.len.limit != 0)

#Calculate % individuals with >= 75% of exon for each exon
col <- col / (length(sample.len[,1]))

#Extract only 75_75 exons from original dataset
length7575<-sample.data[,col>=0.50]

#Get a list of names of 75_75 exons
l_75_75[[i]]<-names(length7575[1,])

}

x<-list(Canellales=l_75_75[[1]],
        Chloranthales=l_75_75[[2]],
        Laurales=l_75_75[[3]],
        Magnoliales=l_75_75[[4]],
        Piperales=l_75_75[[5]])

par(mar=c(1,1,1,1))


library(RColorBrewer)
library("ggVennDiagram")
# Default plot
ggVennDiagram(x,
              label = c("percent"),
              label_alpha = 0.5,
              label_geom = c("label")) + scale_fill_gradient(low=brewer.pal(8,"Greys")[1],high = brewer.pal(8,"Greys")[5])


```

```{r 0_0_tree,echo=FALSE,message=F,warning=F,results='hide',fig.width=15,fig.height=17.5,fig.cap="ASTRAL-III tree annotated with local posterior probability (LPP) and Quartet Support (QS). Inner pie charts on nodes show the value of LPP for the preceding branch (0-1) in black and white. Outer pie charts show QS in grey and white. QS is the proportion of the highest-frequency quartet at each branch i.e. if the QS value is 60% then 60% of gene (supercontig) trees used contain a quartet that matches the relationship shown in the ASTRAL tree. Filtering was done based on exon recovery statistics per locus (> 50% length of exon & > 50% individuals) and sequences representing <10% of the exon were removed from the dataset before alignment. Tree was constructed using 190 loci in total after removing potential paralogs (5). Tip labels are coloured by order."}

par(mar=c(0,0,0,0))

library(ape)
library(phylotools)
library(RColorBrewer)
library(tidyr)
library(ggplot2)

palette(colorRampPalette(brewer.pal(name="Set1", n = 8))(17))

#read in tree and tips
phy <- read.tree("data/trees/astral_r10_l50_i50/astral_bs10_LPP.tre")
df <- read.csv("data/sample_data - samples_for_phylo_AZ.csv")
df <- df[df$Namelist %in% phy$tip.label, ]
df <-df %>% unite("genus_species", Family:Genus:Species, na.rm = TRUE, remove = FALSE)

##tip label change
dfnames <- data.frame(df$Namelist, df$genus_species)
phy <- sub.taxa.label(phy, dfnames)
phy$edge.length[phy$edge.length == "NaN"] <- 0.25

#cols
df <- df[match(phy$tip.label, df$genus_species), ]

#plot tree QS
phy <- root(phy, "Chloranthaceae_Chloranthus_spicatus", edgelabel = T)
phy <- ladderize(phy)
plot(
  phy,
  cex = 0.5,
  label.offset = 0.05,
  align.tip.label = F,
  edge.width = 2,
  edge.color = "grey",
  tip.color = as.numeric(as.factor(df$Order))
)
p <- character(length(phy$node.label))

phy <- read.tree("data/trees/astral_r10_l50_i50/astral_bs10_QS.tre")
nodelabels(
  pie = cbind(as.numeric(phy$node.label), 100 - as.numeric(phy$node.label)),
  piecol = c("grey", "white"),
  cex = 0.25
)

phy <- read.tree("data/trees/astral_r10_l50_i50/astral_bs10_LPP.tre")
nodelabels(
  pie = cbind(as.numeric(phy$node.label), 1 - as.numeric(phy$node.label)),
  piecol = c("black", "white"),
  cex = 0.125
)

legend(
  'bottomleft',
  legend = c(
    'canellales',
    'chloranthales',
    'laurales',
    'magnoliales',
    'piperales',
    'Local PP',
    'Quartet support'
  ),
  text.col = c(1, 2, 3, 4, 5,"black","black"),
  pt.cex = c(NA,NA,NA,NA,NA,2,3),
  pch = c(NA,NA,NA,NA,NA,16,16),
  col = c(NA,NA,NA,NA,NA,"black","grey"),
  bty = 'n'
)

```